### ISSUE TITLE
Please unlock the Sega 32X native rom support on Analogue Mega-SG !

### ISSUE THREAD

**Issue by LRyu2** [analoguejb/Analogue-Mega-Sg-JB#12]
Message to designers, programmers and JB of this great Analogue Mega-SG : since Sergiy Dvodnenko's (@srg320_) work has enabled the full support of rom Sega 32x on MiSTer FPGA, and the Mega-SG has an identical FPGA core, hasn't the time come to unleash the potential of the Mega-SG for the support of Sega 32x ROM ?  And if not, why ?

**Comment by chipperkwah** [comment]
LRyu2, the FPGA core in the Mega SG is not identical to the MISTer Genesis core; it is a completely separate core developed by Kevin Horton and the Analogue engineers. As such, it does not have 32x support that can be "unlocked" in the way you describe.

**Comment by LRyu2** [comment]
Chipperkwah, my description is too simplistic, and I thank you for pointing this out. I'll try to rephrase it.

Starting from the fact that :

-Analogue Mega-SG
-MiSTer FPGA (DE10-Nano based on)

Have an identical CPU : Altera Cyclone V .

This raises the question: is it technically possible for the Mega-SG with a specific software 32X core to boot and interact with the 32X roms/cart ?

In principle, it's not a problem of technical limitations, but of our own will, and it's a real shame that this possibility won't see the light of day.

A 32X blob/core can always be adapted and incorporated on an experimental basis to boot up and play 32X games on Mega-SG, the Sergiy Dvodnenko's (@srg320_) 32X core proves it can be done on MiSTer DE10-Nano.

I'm not asking you to emulate/simulate the other systems available on Cyclone V, but to focus on the ultimate megadrive/genesis.

Unlocking is mostly an idea... And if the answer is no, knowing why is the question ?

**Comment by tjanas** [comment]
This issue should be closed.

**Comment by BooBerry** [comment]
Just noticed this so a bit of a necrobump.

No, it's not possible for the Mega Sg to support a 32X core.

> Unlocking is mostly an idea... And if the answer is no, knowing why is the question ?

It's simple, the FPGA in the Mega Sg isn't powerful enough nor does it have enough logic elements (LEs) to fit the 32X core in addition to the Genesis core at the same time. Plus the FPGA found on the DE-10 Nano board used by MiSTer is way, way more powerful than the Mega Sg's. Remember, it'd have to do both systems at the same time in addition to the analog video mixing the 32X hardware does. Add the fact that the 32X hardware is more powerful than either the Genesis or Sega CD hardware, since it contains two SH-2 CPUs which is the same CPUs the Saturn uses. There's more to it than that (e.g. RAM stuff) and why it's not possible, but for the sake of simplicity let's just use that low hanging fruit of not enough logic elements.

You're thinking about it wrong and making the assumption that the Cyclone V FPGA used in the Mega Sg is the same exact Cyclone V FPGA used on the DE-10 Nano (MiSTer). It's not, not at all, they're different variants of the Cyclone V. The Mega Sg's Cyclone V only has around 49,000 logic elements whereas the Cyclone V on the DE-10 Nano has 110,000 logic elements which is more than double that. The FPGA used in the Mega Sg is cheaper (so less LEs) and the one on the DE-10 Nano is more expensive (so more LEs).

You should do some research on logic elements of a FPGA and how that's pretty much the main limiting factor. It is also why the Super Nt jailbreak SNES core can't support certain special chips like the SuperFX and SA-1. And it's also why the Mega Sg jailbreak can't load Sega CD game images, it's not powerful enough to handle both Genesis and CD cores at the same time.

TL;DR, it's not possible for the Mega Sg to support a 32X core at all, it's not powerful enough. Even if you try to port the 32X MiSTer core it's not going to fit on the Mega Sg, you're going to run out of resources (logic elements).

**Comment by tjanas** [comment]
@LRyu2 please close this issue
